cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/DefaultStore.java

```diff
@@ -987,9 +970,9 @@ private ApplicationSpecificationWithChangedFlows(ApplicationSpecification delega
   }
 
   private static ApplicationSpecification replaceWorkerInAppSpec(final ApplicationSpecification appSpec,
-                                                                 final Id.Program id,
+                                                                 final ProgramId id,
                                                                  final WorkerSpecification workerSpecification) {
-    return new ApplicationSpecificationWithChangedWorkers(appSpec, id.getId(), workerSpecification);
+    return new ApplicationSpecificationWithChangedWorkers(appSpec, id.getEntityName(), workerSpecification);
```

Comment:
its the same as ``getProgram`` for ``ProgramId``, ``getApplication`` for ``ApplicationId``, etc